### ISSUE TITLE
Clipping issue in Print Composer

### ISSUE THREAD

**Issue by DawnBrawn** [qgis/QGIS#42192]
I'm trying to use clip feature with the ESRI:102021 projection, but it looks like some feature aren't rendered after clipping.
QGIS 3.16.4
here's before and after clipping:
![Screenshot (33)](https://user-images.githubusercontent.com/25706516/110754947-f8663180-827a-11eb-9510-61ca0a4f3728.png)
![Screenshot (34)](https://user-images.githubusercontent.com/25706516/110754951-f8fec800-827a-11eb-9a95-993e9419efbd.png)

**Comment by gioman** [comment]
@DawnBrawn Can you attach a sample project (with the layout) and sample data?

**Comment by DawnBrawn** [comment]
[Proyeksi kutub.zip](https://github.com/qgis/QGIS/files/6121499/Proyeksi.kutub.zip)

**Comment by DawnBrawn** [comment]
Hi @gioman , I think I found the issue, although I made the Project's CRS and Map Print Composer's CRS to have ESRI:102021 projection, turns out that the World Map and Grid feature isn't the same projection (still in default EPSG:4326).
So, I reprojection the World Map and Grid into ESRI:102021, and it fixes the problem!
However, I'd like to see if the clip feature doesn't bugged when they are in different projection like in the sample project.
Thank you for the response!
